packages/nextjs/src/sharedTypes/component-props.ts

```diff
@@ -21,27 +21,20 @@ export type ComponentPropsFetchFunction<NextContext, FetchedProps = unknown> = {
 };
 
 /**
- * Shape of getServerSideProps function on component level
+ * Common server component props used in both SSG and SSR modes
```

Comment:
Suggestion of how to improve and extend the description to make it more clear

```suggestion
 * Defines the shape of a data-fetching function used at the component level.
 *
 * This function can be used in both **Server-Side Rendering (SSR)** and **Static Site Generation (SSG)** contexts.
 * It enables component-specific data loading that integrates with Next.js rendering flows.
 *
 * The returned props are passed directly to the component at render time.
```